docs/understand/performance-best-practices.md

```diff
@@ -31,9 +26,61 @@ When accessing data using the branch name (e.g. `lakefs://repo/main/path`) lakeF
 For more information, see [how uncommitted data is managed in lakeFS][representing-refs-and-uncommitted-metadata].
 
 ## Operate directly on the storage
-Sometimes, storage operations can become a bottleneck. For example, when your data pipelines upload many big objects.
+Storage operations can become a bottleneck when operating on large datasets.
+
 In such cases, it can be beneficial to perform only versioning operations on lakeFS, while performing storage reads/writes directly on the object store.
-lakeFS offers multiple ways to do that:
+lakeFS offers multiple ways to do that.
+
+### Use zero-copy import
+To import object into lakeFS, either a single time or regularly, lakeFS offers a [zero-copy import][zero-copy-import] feature.
+Use this feature to import a large number of objects to lakeFS, instead of simply copying them into your repository.
+This feature will create a reference to the existing objects on your bucket and avoids the copy.
+
+The lakeFS blog documents a number of patterns for importing data: [Import Data to lakeFS: Effortless, Fast, and Zero Copy](https://lakefs.io/blog/import-data-lakefs/). 
+
+An import essentially scans the supplied dataset (eg a path in a GCS or S3 bucket) and records various metadata about the files in lakeFS. This is purely a metadata operation, and does not require copying any data.
+
+From that point on, the source data should be considered frozen and immutable.
+
+You can re-import a dataset, and capture any changed or newly added files (this is use case 3 / option 2 in the blog). What that will not do is deal nicely with updated or overwritten files.
+
+This pattern will work nicely for append-only setups.
+
+### Writing directly to the object store
+In addition to importing large, batch-generated datasets, we may want to to add a few new files after an initial import, or to modify existing files. lakeFS allows “uploading” changes to a dataset.
+
+Unlike an import, in the case of an upload, lakeFS is in control of the actual location the file is stored at in the backing object store. This will not modify any directories you may have “imported” the dataset from originally, and you will need to use lakeFS to get consistent views of the data; see the next subsection for advice on scalable reads.
+
+If we need to upload a lot of files, we likely want to avoid writing directly through the lakeFS service. lakeFS allows this, if we follow a particular pattern: we request a location from lakeFS to which a new file should go, and then use regular object store (S3, GCS, etc) client to upload the data directly.
```

Comment:
I didn't test the function but I assume this way is using PreSigned URL to upload the object, and now the function implemented for S3, it will just throw an error for GCS and Azure:
https://github.com/treeverse/lakeFS/blob/eed7c9c35516019c50b42865b5ca8ea99c590ff4/pkg/block/gs/adapter.go#L587-L589
https://github.com/treeverse/lakeFS/blob/eed7c9c35516019c50b42865b5ca8ea99c590ff4/pkg/block/azure/adapter.go#L601-L603

Reply:
I also have not yet tested but it seems to me that `GetPreSignedURL` [is implemented](https://github.com/treeverse/lakeFS/blob/eed7c9c35516019c50b42865b5ca8ea99c590ff4/pkg/block/gs/adapter.go#L132) , it's only the _multipart_ upload that is not.